aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthSession.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.auth.cognito;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.amplifyframework.auth.AuthSession;
+import com.amplifyframework.auth.AuthUserState;
+
+import com.amazonaws.auth.AWSCredentials;
+
+import java.util.Objects;
+
+public final class AWSCognitoAuthSession extends AuthSession {
+    private AWSCredentials awsCredentials;
+    private String identityId;
+    private String accessToken;
+    private String idToken;
+    private String refreshToken;
+
+    public AWSCognitoAuthSession(
+            @NonNull AuthUserState state
+    ) {
+        super(state);
+    }
+
+    public AWSCognitoAuthSession(
+            @NonNull AuthUserState state,
+            @Nullable AWSCredentials awsCredentials,
+            @Nullable String identityId,
+            @Nullable String accessToken,
+            @Nullable String idToken,
+            @Nullable String refreshToken
```

Comment:
If all of these things can be null, how do we get a predictable experience for the user? Right now, they'll have to null check anything they get back from a getter on this object. Can we make fewer of these things null?

Alternately, we could provided `hasFoo()`, methods, to help a little:
```
if (session.hasIdentityId()) {
    String idnetityId = session.getIdnetitytId();
}
```

Or, a functional style:
````
session.ifPresent(SessionToken.REFRESH, /* Action */ () -> {
   // This gets executed if there is a session token.
});
```

Reply:
This is a raw escape hatch data object where the user gets back whatever underlying AWS Credentials may or may not exist depending on the type of account and whether it's signed in or not. We can talk about it more on the call.

Reply:
Add a builder to make it immutable and comment on the reason some things can be null